webpack/assets/javascripts/react_app/components/common/forms/ForemanForm/ForemanForm.js

```diff
@@ -4,12 +4,19 @@ import PropTypes from 'prop-types';
 import Form from '../Form';
 import { translate as __ } from '../../../../common/I18n';
 
-export const prepareErrors = errors =>
-  Object.keys(errors).reduce((memo, key) => {
-    const errorMessages = errors[key];
-    memo[key] = errorMessages ? errorMessages.join(', ') : errorMessages;
-    return memo;
-  }, {});
+export const prepareErrors = exception => {
+  const { errors } = exception;
+  try {
+    return Object.keys(errors).reduce((memo, key) => {
+      const errorMessages = errors[key];
+      memo[key] = errorMessages ? errorMessages.join(', ') : errorMessages;
+      return memo;
+    }, {});
+  } catch (e) {
+    // so we don't swallow non-form-related JS errors
+    throw exception;
```

Comment:
should we also do something with `e`?

Reply:
@MariaAga  The problem I was having was that the Formik errors were an object that looks like `{ errors: { name: "can't be blank" } }`, but regular JS errors could be just a string.  Both were getting passed to `prepareErrors` as the function argument.  Since `prepareErrors` is already inside a `catch` (see line 29), the result was that Formik errors were correctly processed, but other JS errors were just swallowed.  It took me a long time to figure out what was happening.

Now, since I added the `try...catch`, if a regular JS error gets passed to `prepareErrors`, the `try` block will fail because it's not an object shaped like the above, and will pass control to the `catch` block, so that the error is no longer swallowed.

The only time `e` would come into play is if there were an error in the `reduce` in the `try` block, which previously wasn't even handled anyway.  So I don't think it's worth worrying about.